hamcrest/Hamcrest/Type/IsNumeric.php

```diff
@@ -19,10 +19,36 @@ public function __construct()
 
     public function matches($item)
     {
+        if ($this->is_hexadecimal($item)) {
```

Comment:
Will real hexadecimal (not written as a string) pass this check (`matches` or `is_hexadecimal`)?